activation/wire/wire_v2.go

```diff
@@ -222,6 +225,11 @@ func (sp *SubPostV2) Root(prevATXs []types.ATXID) []byte {
 		return nil // invalid index, root cannot be generated
 	}
 	tree.AddLeaf(prevATXs[sp.PrevATXIndex].Bytes())
+
+	var leafIndex [8]byte
+	binary.LittleEndian.PutUint64(leafIndex[:], sp.MembershipLeafIndex)
+	tree.AddLeaf(leafIndex[:])
+
```

Comment:
`leafIndex` needs to be of type `types.Hash32` otherwise encoding / validating proofs becomes a pain (leafs of different sizes in the proof). See also this PR as reference: https://github.com/spacemeshos/go-spacemesh/pull/6043